src/stream.rs

```diff
@@ -317,12 +317,12 @@ pub(crate) fn connect_https(unit: &Unit, hostname: &str) -> Result<Stream, Error
     Ok(Stream::new(https_stream, remote_addr))
 }
 
-/// If successful, returns a `TcpStream` and the remote address it is connected to.
+/// If successful, returns a `Box<dyn ReadWrite>` around a TcpStream and the remote address it is connected to.
```

Comment:
This comment is not strictly accurate, since this could now be wrapping a TcpStream _or_ a TLS stream.